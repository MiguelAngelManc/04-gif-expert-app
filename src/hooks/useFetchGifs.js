import { getGifs } from "../helpers/getGifs";

const { useState, useEffect } = require("react");

export const useFetchGifs = (categoria) =>{
    const [state, setstate] = useState({
        data: [],
        loading: true
    })

    useEffect(()=>{
        getGifs(categoria)
            .then( imgs =>{
                    setstate({
                        data: imgs,
                        loading:false,
                    })
            });
    },[])

    return state;

}

