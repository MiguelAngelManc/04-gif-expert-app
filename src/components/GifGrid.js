import React from 'react'
import PropTypes from 'prop-types'
import GifGridItem from './GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs'

const GifGrid = ({categoria}) => {

    const { data:imagenes, loading } = useFetchGifs(categoria);


    return ( 
        <>
            <h3>{categoria}</h3>

            {loading && <p className="animate__animated animate__flash">Cargando... </p>}

            <div className="card-grid ">
                {imagenes.map(x=><GifGridItem {...x} />)}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    categoria:PropTypes.string.isRequired,
}

export default GifGrid
