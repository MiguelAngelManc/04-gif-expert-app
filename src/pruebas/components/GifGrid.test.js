import { React } from "react";
import { render } from "@testing-library/react";
import GifGrid from "../../components/GifGrid";
import { shallow } from "enzyme";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');


describe('Tests de GifGrid',()=>{

    const categoria = "One punch"
    

    test('Snapshot de GifGrid',()=>{

        useFetchGifs.mockReturnValue({
            data:[],
            loading:true,
        })

        const wrapper = shallow(<GifGrid categoria={categoria} />)

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe mostrar items cuando carga',()=>{

        const mockData = [{
            id: 'ABC',
            url: 'https://blablablalba',
            title: 'Titulo'
        }]

        useFetchGifs.mockReturnValue({
            data:mockData,
            loading:false,
        })

        const wrapper = shallow(<GifGrid categoria={categoria} />)
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(mockData.length)
    })

})