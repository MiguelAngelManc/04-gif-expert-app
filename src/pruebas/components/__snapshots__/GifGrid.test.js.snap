// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests de GifGrid Debe mostrar items cuando carga 1`] = `
<Fragment>
  <h3>
    One punch
  </h3>
  <div
    className="card-grid "
  >
    <GifGridItem
      id="ABC"
      title="Titulo"
      url="https://blablablalba"
    />
  </div>
</Fragment>
`;

exports[`Tests de GifGrid Snapshot de GifGrid 1`] = `
<Fragment>
  <h3>
    One punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando... 
  </p>
  <div
    className="card-grid "
  />
</Fragment>
`;
